bftengine/tests/bcstatetransfer/bcstatetransfer_tests.cpp

```diff
@@ -417,6 +417,7 @@ class BcStTestDelegator {
     return stateTransfer_->checkStructureOfVirtualBlock(virtualBlock, virtualBlockSize, pageSize, logger);
   }
   const Digest& computeDefaultRvbDataDigest() const { return stateTransfer_->computeDefaultRvbDataDigest(); }
+  void enableFetchingState(void) { return stateTransfer_->enableFetchingState(); }
```

Comment:
I think you didnt understand me.
enableFetchingState should be completely removed from priduct code.
define here:
```

void enableFetchingState(void) { 
   DataStoreTransaction::Guard g(stateTransfer_->psd_->beginTransaction());
  g.txn()->setIsFetchingState(true);
 }
```

Reply:
Thanks Gil.
I've worked on all your latest comments and updated the PR.